sdk/core/azure-core/src/environment_log_level_listener.cpp

```diff
@@ -114,7 +115,8 @@ EnvironmentLogLevelListener::GetLogListener()
                   << Azure::DateTime(std::chrono::system_clock::now())
                          .ToString(
                              DateTime::DateFormat::Rfc3339, DateTime::TimeFractionFormat::AllDigits)
-                  << "] " << LogLevelToConsoleString(level) << " : " << message << std::endl;
+                  << " T: " << std::this_thread::get_id() << "] " << LogLevelToConsoleString(level)
+                  << " : " << message << std::endl;
```

Comment:
shouldn't this be an option for the logger?  It might be noisy for people not using multi-thread.  I think it would be goo to opt-in for logging thread info, but to keep it off by default.  

But, this can be done in a later iteration...